client/state/data-layer/wpcom/concierge/shifts/index.js

```diff
@@ -0,0 +1,36 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { noRetry } from 'state/data-layer/wpcom-http/pipeline/retry-on-failure/policies';
+import { fetchConciergeShiftsSuccess, fetchConciergeShiftsFailed } from 'state/concierge/actions';
+import { CONCIERGE_SHIFTS_FETCH } from 'state/action-types';
+
+export const handleFetchConciergeShifts = ( { dispatch }, action ) => {
+	const { scheduleId } = action;
+
+	dispatch(
+		http(
+			{
+				method: 'GET',
+				path: `/concierge/schedules/${ scheduleId }/shifts`,
+				apiNamespace: 'wpcom/v2',
+				retryPolicy: noRetry(),
+			},
+			action
+		)
+	);
+};
+
+export default {
+	[ CONCIERGE_SHIFTS_FETCH ]: [
+		dispatchRequest(
+			handleFetchConciergeShifts,
+			fetchConciergeShiftsSuccess,
```

Comment:
See comments above with @dmsnell (we may want to ditch special success/error actions) but if we do want to keep them, you'll need to specifically dispatch them. This newer `dispatchRequest` helper doesn't auto-dispatch anything, so you can't just pass in an action, you need to create a special handler that actually dispatches the action.